### PR TITLE
Add build with mutual file

### PR DESCRIPTION
**Pull request by Melanitski**
For rails project with bower, we need to have one main file.
When we use `rake assets:precompile` our application compress file that marked as main in bower file. 

I thinks we need to have one main file with validation and rules or extract validation-rules to other bower component

### CODE REVIEW

**Comment by huei90** [comment]
I'm not really understand.

Would you mind describe more about this change ? Why name mutual and why change the bower name.

Thanks for your PR.

**Comment by Melanitski** [comment]
Main goal is merge 2 build files of repository. Now `main` attribute in bower.js config references to complete solution 

Now repository has 2 base files and we couldn't use it with assets pipeline

File: bower.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "huei90"
   ],
   "description": "Client-side Validation for AngularJS",
-  "main": "dist/angular-validation.js",
+  "main": "dist/angular-validation-mutual.js",
```
Comment (huei90) [inline bower.json]: Is it possible not to change the file name? @Melanitski

**Comment by huei90** [comment]
Hi @Melanitski May I know what's the `mutual` means this?

**Comment by austinpray** [comment]
Mutual is not a good name. "Bundle" or "bundled" would be more appropriate.

**Comment by huei90** [comment]
Hi @Melanitski Any update on this? Else I'm going to close or you can give us a better suggestion on `mutual` =)

**Comment by chriszamierowski** [comment]
@huei90 Having the same issue. The problem is that currently ```bower.json``` only has ```dist/angular-validation.js``` listed in [main](https://github.com/huei90/angular-validation/blob/master/bower.json#L9), but it should list "The primary acting files necessary to use your package" (from [Bower docs](http://bower.io/docs/creating-packages/)).

The problem that it's causing is that certain Rails configurations use ```main``` to automatically include files (as I'm sure others do as well), but doing so currently won't include ```dist/angular-validation-rule.js```. 

So one solution would be to include both files in ```main```: 
```javascript
"main": ["dist/angular-validation.js", "dist/angular-validation-rule.js"]
```

This way both necessary files will be included whenever automatically including this bower package.

Another solution (probably the preferred) would be to combine both ```dist/angular-validation.js``` and ```dist/angular-validation-rule.js``` into one file for distribution, which I believe is what @Melanitski was getting at. Is there a reason to keep these files separate?

**Comment by Melanitski** [comment]
I thinks, that @chriszamierowski propose a good solution.
Better to reject this pull request and add one line to bower.file

**Comment by huei90** [comment]
Thanks @austinpray @Melanitski , finally I got the point now.

But I don't suggest you to use `angular-validation-rule.js` because I'm using that as an example on using and defining the validation. It's better if you can create your own config file for the validation rule.

Hope this is fine or anything I can help on improvement? Please let me know.

**Comment by Melanitski** [comment]
In source of `angular-validation-rule.js` we have some trivial solutions and we use it for saving time and remove duplication

**Comment by chriszamierowski** [comment]
@huei90 ah, makes sense. I was just following the setup steps and didn't notice that file was as simple as it was. Thanks

**Comment by huei90** [comment]
:smile:  Since I didn't focus on building the type of validation, I though every project has their own validation purpose, that's why I don't suggest use my rule as your project, you should build it yourself for your own purpose =)

**Comment by huei90** [comment]
Close it. Open if needed.

**Pull request closed without merging**